lib/transport/ip/http-event-parser.js

```diff
@@ -79,21 +81,39 @@ class HttpEventParser extends EventEmitter {
           break;
         }
         case State.HEADERS_COMPLETE: {
-          const contentLength = parseInt(this.headers['content-length'], 10);
-          const toCopy = Math.min(
-            contentLength - this.body.length,
-            this._pending.length
-          );
-
-          this.body = Buffer.concat([
-            this.body,
-            this._pending.slice(0, toCopy),
-          ]);
-          this._pending = this._pending.slice(toCopy, this._pending.length);
-
-          if (this.body.length === contentLength && this.protocol === 'EVENT') {
-            this.emit('event', this.body);
-            this._reset();
+          if (typeof this.headers['content-length'] !== 'undefined') {
+            const contentLength = parseInt(this.headers['content-length'], 10);
+            const toCopy = Math.min(
+              contentLength - this.body.length,
+              this._pending.length
+            );
+            this.body = Buffer.concat([
+              this.body,
+              this._pending.slice(0, toCopy),
+            ]);
+            this._pending = this._pending.slice(toCopy, this._pending.length);
+            if (this.body.length === contentLength &&
+                this.protocol === 'EVENT') {
+              this.emit('event', this.body);
+              this._reset();
+            }
+          } else if (typeof this.headers['content-type'] !== 'undefined') {
+            if (this._pending.indexOf(
+              EVENT_SUFFIX,
+              this._pending.length - EVENT_SUFFIX.length
+            ) !== -1) {
+              let body = this._pending.toString('utf8');
+              const firstPosition = body.indexOf('{');
+              const lastPosition = body.lastIndexOf('}');
+              body = body.substr(
+                firstPosition,
+                lastPosition - firstPosition + 1
+              );
+              this.emit('event', body);
+              this._pending = Buffer.alloc(0);
```

Comment:
`this._pending` should be set to the remaining content, if any, rather than just reset.